packages/babel-plugin-transform-react-jsx-source/src/index.js

```diff
@@ -0,0 +1,56 @@
+
+ /**
+ * This adds {fileName, lineNumber} annotations to React component definitions
+ * and to jsx tag literals.
+ *
+ *
+ * == JSX Literals ==
+ *
+ * <sometag />
+ *
+ * becomes:
+ *
+ * <sometag __source={{fileName: 'this/file.js', lineNumber: 10}}/>
```

Comment:
I wonder if it'd be better to use a string ending in `:column:line` just like stack traces are.

```js
<sometag __source="this/file.js:10:0"/>
```

It follows the stack frame convention, is smaller and the string gets interned without the need to hoist the object. So avoids the need for further allocations.